### ISSUE TITLE
[Feature?] Mixed shallow / deep message format

### ISSUE THREAD

**Issue by rvantonisse-accessibility** [kaisermann/svelte-i18n#129]
**Is your feature request related to a problem? Please describe.**
From the docs:
> If the message id literal value is not in the root of the dicitonary, . (dots) are interpreted as a path:
>
> ```json
> {
>   "shallow.prop": "Shallow property",
>   "deep": {
>     "property": "Deep property"
>   }
> }
> ```

I have a mixed format for convenience which I want to use:
**EDIT**: Decided just to show my actual usecase since it is more clarifying!
```json
{
  "WCAG": {
      "SUCCESS_CRITERION": {
          "1.1.1": {},
          "1.2.1": {},
          "1.3.1": {},
      },
  }
}
```

**Describe the solution you'd like**
I would like to be able to use mixed deep / shallow / deep formatting messages like I have shown in my mixed format example:
```json
{
  "WCAG": {
      "SUCCESS_CRITERION": {
          "1.1.1": {},
          "1.2.1": {},
          "1.3.1": {},
      },
  }
}
```

**Describe alternatives you've considered**
I can access this with the newer `$json` store, but I really like to use `$format` in this case on component level instead of `$json` at parent component level.

I could do:
```json
{
  "WCAG": {
      "SUCCESS_CRITERION": {
          "1": {
              "1": {
                  "1": {}
              }
          },
      },
  }
}
```

But I have 78 deepest properties like that, and growing. (Specificly talking about WCAG succes criteria translations here)

**How important is this feature to you?**
Using `$json` at parent component level seems ok for now, but the structure off my application's UI might change in a way that this solution is not feasible anymore and acces to translations at component level are preferred.

**Additional context**
This is the component in question: a WCAG Success criterion.
![image](https://user-images.githubusercontent.com/60504106/107941104-80616000-6f89-11eb-8435-bf5f0f6168de.png)

See future code at: https://github.com/AccessibilityNL/wcag-em-report-tool/tree/rvantonisse/feature/rebuild-svelte-wcag-data

See a live version of app (not up to date, but to get the picture): https://accessibilitynl.github.io/wcag-em-report-tool/svelte-preview/

**Comment by rvantonisse-accessibility** [comment]
I just have been implementing component level translating with $json like this:

```javascript
$: TRANSLATED_THING = $json('DEEP.SHALLOW.PROP')[index];
```
Feels weird that I just cant use it like:
```javascript
$: TRANSLATED_THING = $format(`DEEP.SHALLOW.PROP.${index}`);
```

But still gets the job done, although I am not sure if this is the intended / right way.

**Comment by kaisermann** [comment]
Hey, @rvantonisse-accessibility 👋 The idea of the `format` method is to resolve to a string. That's exactly why the `json` method exist.

**Comment by kaisermann** [comment]
Hey @rvantonisse-accessibility 👋 

I started thinking that the `[index]` way was the correct one. However, it came to my mind that if we support string with dots as (shallow) keys, we might as well support them for deep properties as well. There's a processing cost here, but shouldn't be that bad.

**Comment by kaisermann** [comment]
Released in `v3.3.5` 🎉

**Comment by rvantonisse-accessibility** [comment]
Hey @kaisermann , 

this is great. Thank you for your work!
